pkg/sql/backfill.go

```diff
@@ -843,10 +841,16 @@ func (sc *SchemaChanger) distBackfill(
 						otherTableDescs = append(otherTableDescs, *table.TableDesc())
 					}
 				}
-				rw := &errOnlyResultWriter{}
+				metaFn := func(_ context.Context, meta *distsqlpb.ProducerMetadata) {
+					if meta.BulkProcessorProgress != nil {
```

Comment:
I think we avoid wrapping todoSpans access in a mutex because all the access from the callback is from the same goroutine (the distsql inbound goroutine) and all happens while this goroutine is blocked on `Run`.

If that is correct, that's a _slightly_ brittle assumption but it works for me for now.